subchannel_test.go

```diff
@@ -104,3 +109,92 @@ func TestAddReusesPeers(t *testing.T) {
 		assertHaveSameRef(t, set.main, set.isolated)
 	})
 }
+
+func TestSetHandler(t *testing.T) {
+	// Generate a Handler that expects only the given methods to be called.
+	genHandler := func(methods ...string) tchannel.Handler {
+		allowedMethods := make(map[string]struct{}, len(methods))
+		for _, m := range methods {
+			allowedMethods[m] = struct{}{}
+		}
+
+		return tchannel.HandlerFunc(func(ctx context.Context, call *tchannel.InboundCall) {
+			method := call.MethodString()
+			assert.Contains(t, allowedMethods, method, "unexpected call to %q", method)
+			err := raw.WriteResponse(call.Response(), &raw.Res{Arg3: []byte(method)})
+			require.NoError(t, err)
+		})
+	}
+
+	ch := testutils.NewServer(t, testutils.NewOpts().
+		AddLogFilter("Couldn't find handler", 1, "serviceName", "svc2", "method", "bar"),
```

Comment:
nit: trailing comma here is a little strange, usually i just put the trailing `)` on this line too.